### PR TITLE
Blur book covers with content_warnings when safe-mode is enabled

### PR DESCRIPTION
**Pull request by rs1dev**
Follow-Up to PR #7775 and Issue #7416 (part 2) to blur covers containing nsfw content.

This PR implements the blurring functionality of the opt-in safe-mode feature. Books tagged as nsfw/containing inappropriate content via a `content_warnings:*` subject will have their covers blurred if the user has enabled Safe Mode in 'Manage Privacy and Content Moderation Settings' on the Settings page. 

### Technical

When Safe Mode is enabled, an `sfw` cookie is set to 'yes' and a list of subjects is retrieved  when a search is entered. If a `content_warnings:*` subject is present for that book, a blurred book cover appears else the original book cover is displayed. This also works for the small book previews as shown in the screenshots. 

### Testing

1. Go to `conf/openlibrary.yml` and change line 18 `coverstore_url: http://covers:7075` to `coverstore_url: covers.openlibrary.org` 
2. Remake css: Enter `docker compose run --rm home make css` in the terminal after 'docker compose up' 
3. Restart memcached in docker: `docker compose restart memcached` 
4. Enable safe mode (from Privacy and Content Moderation Settings) 
5. Search for a book (type the book title or any search term) and press Enter. Currently, any book with a cover or preview will be blurred for testing purposes, but this will be adjusted for only books with the `content-warnings:*` subject. 
7. On the search results page on the top right, just above the displayed book components, uncheck the 'Solr Editions Beta' checkbox. 
8. The book cover and/or preview should be blurred. 

### Screenshots

bookcover_blur

preview_blur

### CODE REVIEW

**Comment by SivanC** [comment]
@mek @jimchamp I think the second part of the feature is complete! I have done some light testing and so far it looks like everything is working as expected, but we would appreciate any/all feedback you guys have or anyone else has on the implementation/any other issues that arise (no pressure to look at it before our presentation on Monday as that's already planned out :) ). Thank you!

**Comment by SivanC** [comment]
I have found an pretty obvious issue (oops) that the template tries to access the list of subjects even if they aren't grabbed due to the user not being in safe mode, but I have fixed that on my local repo and will push a solution soon.

File: openlibrary/macros/SearchResultsWork.html [hunk 1]
```diff
@@ -87,7 +88,8 @@ <h3 itemprop="name" class="booktitle">
           $if doc.get('ia'):
             &mdash; $_('%s previewable', len(doc.get('ia')))
             $if len(doc.get('ia')) > 1:
-              <span class="preview-covers">
+              $ preview_cover_class = "safe-mode-preview-covers" if blur else "preview-covers"
+              <span class=$preview_cover_class>
```
Comment (mekarpeles) [inline openlibrary/macros/SearchResultsWork.html:92]: Where does this `span` end?

File: openlibrary/plugins/worksearch/code.py [hunk 2]
```diff
@@ -287,14 +287,15 @@ def do_search(
     :param sort: csv sort ordering
     :param spellcheck_count: Not really used; should probably drop
     """
+
+    if web.cookies(sfw="").sfw == 'yes':
+        fields = list(
+            WorkSearchScheme.default_fetched_fields | {'editions'} | {'subject_key'}
```
Comment (mekarpeles) [inline openlibrary/plugins/worksearch/code.py:293]: @jimchamp, here we pass in a special directive to `solr` to get the `subject_key` of editions. And based on this, we can see if that book has a content warning and can then blur the image if the patron has "safe for work" mode enabled in their cookies.

**Comment by jimchamp** [comment]
@SivanC and @rs1dev, sorry for the delay!  As I was reviewing this PR, I noticed that you used the same branch as what was used for the previous PR.  This means that the previous 19 commits will be merged into `master` again with this PR.

I think that the easiest way to fix this is to create a new branch from `master`, add the new commits to the branch using `cherry-pick`, pushing the branch, and creating a new PR.  The following commands should get you most of the way there, you'll just have to open the PR:
```
git checkout master
git pull upstream master
git checkout -b 7416/feature/blur-nsfw-images
git cherry-pick 246376db7fafa28ab35dd05426b990f677d047f4..6e4e24fa7124cc647d544f0732f9e465b6a44029
git push origin HEAD
```

**Pull request closed without merging**